datastore/db/bytebase/FertiScan/new_inspection_function.sql

```diff
@@ -0,0 +1,310 @@
+    END IF;   
+	INSERT INTO organization_information (name,website,phone_number,location_id)
+	VALUES (
+	        input_json->'company'->>'name',
+            input_json->'company'->>'website',
+            input_json->'company'->>'phone_number',
+            location_id
+	)
+	RETURNING id INTO company_id;
+	
+	-- Update input_json with company_id
+	input_json := jsonb_set(input_json, '{company,id}', to_jsonb(company_id));
+
+-- COMPANY END
+
+-- MANUFACTURER
+	-- Check if company location exists by address
+    SELECT id INTO location_id
+    FROM location
+    WHERE address ILIKE input_json->'manufacturer'->>'address'
+	LIMIT 1;
+   
+	IF location_id IS NULL THEN 
+        INSERT INTO location (address)
+        VALUES (
+            input_json->'manufacturer'->>'address'
+        )
+        RETURNING id INTO location_id;
+    END IF;
+	INSERT INTO organization_information (name,website,phone_number,location_id)
+	VALUES (
+	        input_json->'manufacturer'->>'name',
+            input_json->'manufacturer'->>'website',
+            input_json->'manufacturer'->>'phone_number',
+            location_id
+	)
+	RETURNING id INTO manufacturer_id;
+	
+	-- Update input_json with company_id
+	input_json := jsonb_set(input_json, '{manufacturer,id}', to_jsonb(manufacturer_id));
+-- Manufacturer end
+
+-- LABEL INFORMATION
+    INSERT INTO label_information (
+        lot_number, npk, registration_number, n, p, k, company_info_id, manufacturer_info_id
+    ) VALUES (
+        input_json->'product'->>'lot_number',
+        input_json->'product'->>'npk',
+        input_json->'product'->>'registration_number',
+        (input_json->'product'->>'n')::float,
+        (input_json->'product'->>'p')::float,
+        (input_json->'product'->>'k')::float,
+		company_id,
+		manufacturer_id
+    )
+    RETURNING id INTO label_id;
+		
+	-- Update input_json with company_id
+	input_json := jsonb_set(input_json, '{product,id}', to_jsonb(label_id));
+
+--LABEL END
+
+--WEIGHT
+	-- Loop through each element in the 'weight' array
+    FOR record IN SELECT * FROM jsonb_array_elements(input_json->'weight')
+    LOOP
+        -- Extract the value and unit from the current weight record
+        read_value := record->>'value';
+        read_unit := record->>'unit';
+
+        -- Convert the weight value to float
+        value_float := read_value::float;
+       
+       -- Check if the weight_unit exists in the unit table
+	    SELECT id INTO unit_id FROM unit WHERE unit = read_unit;
+	   
+	   -- If unit_id is null, the unit does not exist
+	    IF unit_id IS NULL THEN
+	        -- Insert the new unit
+	        INSERT INTO unit (unit, to_si_unit)
+	        VALUES (read_unit, null) -- Adjust to_si_unit value as necessary
+	        RETURNING id INTO unit_id;
+	    END IF;
+	   
+	   	 -- Insert into metric for weight
+	    INSERT INTO metric (value, unit_id, edited,metric_type_id,label_id)
+	    VALUES (value_float, unit_id, FALSE,'weight'::metric_type,label_id);
```

Comment:
I would use `"fertiscan_0.0.9".metric_type` just to be sure but it seems to be working without the schema prefix.